java/src/jmri/jmrit/logixng/expressions/ExpressionTransit.java

```diff
@@ -0,0 +1,238 @@
+//                 break;
+//
+//             case ValueMatches:
+//                 String blockValue = _selectBlockValue.evaluateValue(conditionalNG);
+//                 currentState = blockValue.equals(block.getValue())
+//                         ? BlockState.ValueMatches.getID() : 0;
+//                 break;
+//
+//             default:
+//                 break;
+//         }
+
+        if (_is_IsNot == Is_IsNot_Enum.Is) {
+            return currentState == checkTransitState.getID();
+        } else {
+            return currentState != checkTransitState.getID();
+        }
+    }
+
+    @Override
+    public FemaleSocket getChild(int index) throws IllegalArgumentException, UnsupportedOperationException {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
+    @Override
+    public int getChildCount() {
+        return 0;
+    }
+
+    @Override
+    public String getShortDescription(Locale locale) {
+        return Bundle.getMessage(locale, "Transit_Short");
+    }
+
+    @Override
+    public String getLongDescription(Locale locale) {
+        String namedBean = _selectNamedBean.getDescription(locale);
+        String state;
+
+        switch (_selectEnum.getAddressing()) {
+            case Direct:
+                TransitState transitState = _selectEnum.getEnum();
+                state = Bundle.getMessage(locale, "AddressByDirect", transitState._text);
+                break;
+
+            case Reference:
+            case LocalVariable:
+            case Formula:
+                state = _selectEnum.getDescription(locale);
+                break;
+
+            default:
+                throw new IllegalArgumentException("invalid _stateAddressing state: " + _selectEnum.getAddressing().name());
+        }
+
+
+        return Bundle.getMessage(locale, "Transit_Long", namedBean, _is_IsNot.toString(), state);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void setup() {
+        // Do nothing
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void registerListenersForThisClass() {
+        if (!_listenersAreRegistered) {
+            _selectNamedBean.addPropertyChangeListener(this);
+            _selectNamedBean.registerListeners();
+            _listenersAreRegistered = true;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void unregisterListenersForThisClass() {
+        if (_listenersAreRegistered) {
+            _selectNamedBean.removePropertyChangeListener(this);
+            _selectNamedBean.unregisterListeners();
+            _listenersAreRegistered = false;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
+        getConditionalNG().execute();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void disposeMe() {
+    }
+
+    public enum TransitState {
+        Idle(2, Bundle.getMessage("Transit_StateIdle")),
```

Comment:
I recommend that you use `Transit.IDLE` instead of `2` and `Transit.ASSIGNED` instead of `4`. It makes the code easier to read and less prone to errors.

Reply:
Changed.